tools/pydfu.py

```diff
@@ -601,7 +622,18 @@ def main():
         command_run = True
 
     if args.path:
-        elements = read_dfu_file(args.path)
+        if str(args.path).endswith(".bin"):
+            if not args.address:
+                raise ValueError("Address must be specified using -a when uploading binary")
+            addr = str(args.address)
+            if addr.startswith("0x") or addr.startswith("0X"):
+                addr = int(addr, 16)
+            else:
+                addr = int(addr)
+            elements = read_bin_file(args.path, addr)
```

Comment:
This logic can be simplified if the address argument is specified as above.

Reply:
Solved by ab0266f1df631264d8edb4f243df18d21035b3f9